build/build_variables.bzl

```diff
@@ -0,0 +1,467 @@
+# WARNING: the contents of this file must BOTH be valid Starlark (for Buck) as well as
+# valid Python (for our cmake build). This means that load() directives are not allowed
+# (as they are not recognized by Python). If you want to fix this, figure out how run
+# this file from cmake with a proper Starlark interpreter as part of the default OSS
+# build process. If you need some nontrivial Starlark features, make a separate bzl
+# file. (Remember that bzl files are not exported via ShipIt by default, so you may also
+# need to update ExecuTorch's ShipIt config.)
+
```

Comment:
Please add a top-level comment explaining what this file contains, the semantics of a given list, and how authors should manage the set of entries. It'd also be helpful to point out the function in Codegen.cmake that needs to be updated when adding a new list to this file.